lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp

```diff
@@ -841,15 +841,15 @@ bool TypeLoweringVisitor::visitStmt(StrictConnectOp op) {
   for (const auto &field : llvm::enumerate(fields)) {
     Value src = getSubWhatever(op.getSrc(), field.index());
     Value dest = getSubWhatever(op.getDest(), field.index());
-    if (field.value().isOutput)
+    if (field.value().isOutput && !op.getDest().getType().isa<RefType>())
```

Comment:
I think field is correct for flow, so if you have a ref in a bundle with flip you need to do this swap.  And if you don't then it shouldn't ever happen anyway?

Reply:
Ah, this also is restoring behavior modified with RefConnect unexpectedly/untested.

Agreed that flip here makes sense, sorry about that.  Don't think matters presently/yet but will soon and better to have this right anyway :+1:.  Will dig and see why this was even added, offhand not sure how it's reachable.

(.. Is this checking if the destination is a reference, so this would be a flip within that reftype? o_O I'll investigate, apologies if my assessment above is incorrect, thanks for calling attention to this however it's resolved!)

Reply:
(restored previous behavior, 9ba2e25f6a0489b31a372c4c24380cc3a08cab99)